runtime/cudaq/platform/default/rest_server/helpers/server_impl/CMakeLists.txt

```diff
@@ -6,26 +6,10 @@
 # the terms of the Apache License 2.0 which accompanies this distribution.     #
 # ============================================================================ #
 
-include(FetchContent)
-FetchContent_Declare(
-    asio
-    GIT_REPOSITORY    https://github.com/chriskohlhoff/asio.git
-    GIT_TAG           asio-1-28-2
```

Comment:
We may want to track if / when to update the submodules regularly... (outside the scope of this PR)